sn_node/src/node/dkg.rs

```diff
@@ -0,0 +1,362 @@
+// Copyright 2022 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::node::{Error, Result};
+use ed25519::Signature;
+use ed25519_dalek::{Keypair, Verifier};
+
+use sn_interface::{
+    messaging::system::DkgSessionId,
+    network_knowledge::threshold,
+    types::{
+        self,
+        keys::ed25519::{pub_key, Digest256},
+    },
+};
+
+use bls::{PublicKey as BlsPublicKey, PublicKeySet, SecretKey as BlsSecretKey, SecretKeyShare};
+use sn_sdkg::{DkgSignedVote, DkgState, NodeId, VoteResponse};
+use std::collections::{BTreeMap, BTreeSet, HashMap};
+use std::sync::Arc;
+use std::time::Instant;
+use xor_name::XorName;
+
+pub(crate) type DkgPubKeys = BTreeMap<XorName, (BlsPublicKey, Signature)>;
+
+pub(crate) struct DkgEphemeralKeys {
+    secret_key: BlsSecretKey,
+    pub_keys: DkgPubKeys,
+}
```

Comment:
Why do we store the `pub_keys` as a map? I'd imagine this struct would be:
```suggestion
pub(crate) struct DkgEphemeralKeys {
    owner: XorName,
    secret_key: BlsSecretKey,
    signature: Signature
}
```
You don't need to hold the public key since it can be re-computed from the secret_key with `secret_key.public_key()`